src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnBomTool.groovy

```diff
@@ -55,41 +80,150 @@ class YarnBomTool extends BomTool implements NestedBomTool<BomToolSearchResult>
-            return true;
+            this.searchResult = searchResult
+            return true
         }
 
-        return false;
+        return false
     }
 
-    public List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
-        final File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory, 'yarn.lock')
-        final List<String> yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
-        final DependencyGraph dependencyGraph = yarnPackager.parse(yarnLockText)
-        final ExternalId externalId = externalIdFactory.createPathExternalId(Forge.NPM, searchResult.searchedDirectory.canonicalPath)
-        final def detectCodeLocation = new DetectCodeLocation.Builder(getBomToolType(), searchResult.searchedDirectory.canonicalPath, externalId, dependencyGraph).build()
+    List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
+        DependencyGraph dependencyGraph
+        def detectCodeLocation
+
+        File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory.canonicalPath, 'yarn.lock')
+        yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
+        yarnExePath = findExecutablePath(ExecutableType.YARN, true, detectConfiguration.getYarnPath())
+
+        if (detectConfiguration.yarnProductionDependenciesOnly) {
+            dependencyGraph = parseYarnList(readYarnListAsLines(yarnExePath))
+        } else {
+            dependencyGraph = yarnPackager.parseYarnLock(yarnLockText)
+        }
+
+        ExternalId externalId = externalIdFactory.createPathExternalId(Forge.NPM, searchResult.searchedDirectory.canonicalPath)
+        detectCodeLocation = new DetectCodeLocation.Builder(getBomToolType(), searchResult.searchedDirectory.canonicalPath, externalId, dependencyGraph).build()
 
         return [detectCodeLocation]
     }
 
-    public List<DetectCodeLocation> extractDetectCodeLocations() {
+    List<String> readYarnListAsLines(String yarnExePath) {
+        File yarnListOutputFile = detectFileManager.createFile(BomToolType.YARN, OUTPUT_FILE)
+        File yarnListErrorFile = detectFileManager.createFile(BomToolType.YARN, ERROR_FILE)
+
+        def exeArgs = ['list', '-prod']
+
+        Executable yarnListExe = new Executable(new File(sourcePath), yarnExePath, exeArgs)
+        executableRunner.executeToFile(yarnListExe, yarnListOutputFile, yarnListErrorFile)
+
+        if (!(yarnListOutputFile.length() > 0)) {
+            if (yarnListErrorFile.length() > 0) {
+                logger.error("Error when running yarn list --prod command")
+                logger.debug(yarnListErrorFile.text)
+            } else {
+                logger.warn("Nothing returned from yarn list --prod command")
+            }
+        }
+
+        yarnListOutputFile.readLines()
+
+    }
+
+    DependencyGraph parseYarnList(List yarnListAsList) {
```

Comment:
This is done similarly to the yarn lock parsing, with several layers and this returns the dependency graph